x/crosschain/migrations/v4/migrate.go

```diff
@@ -159,3 +169,21 @@ func MoveTssToObserverModule(ctx sdk.Context,
 		observerKeeper.SetTSS(ctx, tss)
 	}
 }
+
+// SetBitcoinFinalizedInbound sets the finalized inbound for bitcoin chains to prevent new ballots from being created with same intxhash
+func SetBitcoinFinalizedInbound(ctx sdk.Context, crosschainKeeper crosschainKeeper) {
+	for _, cctx := range crosschainKeeper.GetAllCrossChainTx(ctx) {
+		if cctx.InboundTxParams != nil {
+			// check if bitcoin inbound
+			if common.IsBitcoinChain(cctx.InboundTxParams.SenderChainId) {
```

Comment:
Can we remove this check and do it for all cctxs now that we have adding the finalized indexes?

Reply:
We can't because we don't have the even index saved. For Bitcoin it is always `0`, and it is the only ballot recreated issue.